code/game/objects/structures/shower.dm

```diff
@@ -252,3 +274,28 @@
 	layer = FLY_LAYER
 	anchored = TRUE
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+
+/obj/item/showerparts
+	name = "shower parts"
+	desc = "All the things you need to construct a shower, comes with in-built H2O atom assembler to compensate for Nanotrasen's lack of plumbing installation."
+	icon = 'icons/obj/watercloset.dmi'
+	icon_state = "showerbuild1"
+	var/created_shower_type = /obj/machinery/shower
+
+/obj/item/showerparts/attack_self(mob/user)
+	. = ..()
+	user.visible_message("<span class='danger'>[user] starts putting together a shower.</span>", \
+	"<span class='userdanger'>You start putting together a shower!</span>")
+	if(do_after(user, 40))
```

Comment:
`target = user` so you can progress bar?

Reply:
will do cpn